File: lecture_010(RecursionwithArraylists)/l_010_printmazepathwithjumps.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

void printmazepathwithjumps(int sr, int sc, int dr, int dc, string path) {
    if(sr==dr && sc==dc) {
        cout<<path<<endl;
        return;
    }
    for(int ms=1; ms<=dc-sc; ms++) {
        printmazepathwithjumps(sr, sc + ms, dr, dc, path + "h" + to_string(ms));
    }
    for(int ms=1; ms<=dr-sr; ms++) {
        printmazepathwithjumps(sr + ms, sc, dr, dc, path + "v" + to_string(ms));
    }
    for(int ms=1; ms<=dc-sc && ms<=dr-sr; ms++) {
        printmazepathwithjumps(sr + ms, sc + ms, dr, dc, path + "d" + to_string(ms));
    }
}
int main() {
    int n,m;
    cin>>n>>m;
    printmazepathwithjumps(1,1,n,m,"");
}

File: lecture_010(RecursionwithArraylists)/l_010_printmazepathwithjumps.java
import java.util.Scanner;
public class l_010_printmazepathwithjumps {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int m = scn.nextInt();
        printMazePathWithJumps(1,1,n,m,"");
    }
    public static void printMazePathWithJumps(int sr, int sc, int dr, int dc, String path) {
        if(sr==dr && sc==dc) {
            System.out.println(path);
            return;
        }
        for(int ms=1; ms<=dc-sc; ms++) {
        	printMazePathWithJumps(sr, sc + ms, dr, dc, path + "h" + ms);
        }
        for(int ms=1; ms<=dr-sr; ms++) {
            printMazePathWithJumps(sr + ms, sc, dr, dc, path + "v" + ms);
        }
        for(int ms=1; ms<=dc-sc && ms<=dr-sr; ms++) {
            printMazePathWithJumps(sr + ms, sc + ms, dr, dc, path + "d" + ms);
        }
    }
}